src/Ssh/Ssh/Common/BastionUtils.cs

```diff
@@ -0,0 +1,318 @@
+            }
+
+            try
+            {
+                TunnelServer tunnel = new TunnelServer(context, port, bastion, bastionEndPoint, vmSubscriptionID, bastionPort);
+
+                Thread tunnelThread = new Thread(() => tunnel.StartServer());
+                tunnelThread.Start();
+
+                sshProcess.Start();
+                sshProcess.WaitForExit();
+                int sshExitCode = sshProcess.ExitCode;
+                return sshExitCode;
+            }
+            catch (Exception ex)
+            {
+                string error = "Error connecting to Bastion Host: " + ex.Message;
+                throw new AzPSCloudException(error);
+            }
+
+        }
+
+        public BastionHost FetchDeveloperBastion(string resourceGroupName, string name)
+        {
+            try
+            {
+                var bastion = this.BastionClient.Get(resourceGroupName, name);
+                if (bastion.Sku.Name != "Developer")
+                {
+                    throw new InvalidOperationException("SSH to Bastion host is only support for Developer Bastion Skus");
+                }
+
+                return bastion;
+            }
+            catch (Exception ex)
+            {
+
+                throw new AzPSCloudException($"An error occurred while fetching bastion host: {ex.Message}");
+            }
+        }
+
+        public BastionHost CreateDeveloperBastion(string resourceGroupName, string bastionName, string location, string vNetId)
+        {
+            var virtualNetwork = new SubResource
+            {
+                Id = vNetId
+            };
+
+            var sku = new Sku
+            {
+                Name = "Developer"
+            };
+
+            var bastion = new BastionHost
+            {
+                Location = location,
+                IpConfigurations = new List<BastionHostIPConfiguration>(),
+                VirtualNetwork = virtualNetwork,
+                ScaleUnits = 2,
+                DisableCopyPaste = null,
+                EnableFileCopy = null,
+                EnableIpConnect = null,
+                EnableShareableLink = null,
+                EnableTunneling = null,
+                EnableKerberos = null,
+                EnableSessionRecording = null,
+                Sku = sku,
+                Tags = new Dictionary<string, string>()
+            };
+
+            try
+            {
+                this.BastionClient.CreateOrUpdate(resourceGroupName, bastionName, bastion);
+
+            }
+            catch (Exception ex)
+            {
+                throw new AzPSCloudException($"Error creating bastion: {ex.Message}");
+            }
+            return this.BastionClient.Get(resourceGroupName, bastionName);
+        }
+
+        protected void CheckValidBastionDeveloperLocation(string location)
+        {
+            string[] validLocations = { "centraluseuap", "eastus2euap", "westus", "northeurope", "northcentralus", "westcentralus" };
```

Comment:
Can we check the bastion module for valid regions?

Reply:
The bastion module unfortunately does not have these preset regions, after talking to the bastion team they want to roll out the next few regions as they come along so they want to make this bastion parameter as preview